sim-lib/src/lib.rs

```diff
@@ -290,11 +298,19 @@ struct Payment {
 
 impl Display for Payment {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let dispatch_time = self
+            .dispatch_time
+            .duration_since(UNIX_EPOCH)
+            .expect(&format!(
+                "Failed to compute duration since unix epoch: {:?}",
+                UNIX_EPOCH
```

Comment:
nit: `UNIX_EPOCH` doesn't really need to be included in the panic message because it's a constant.